test/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+aux_source_directory(. TEST_DIR)
+aux_source_directory(../src TEST_DIR)
```

Comment:
really don't like this one...
should do add_library(texas_core  ${SOURCE}) in src Cmake.  
And then target_link_libraries(texas_test texas_core)